language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/ContractMetadata.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.javaapi.data;
+
+import com.daml.ledger.api.v1.ContractMetadataOuterClass;
+import com.google.protobuf.ByteString;
+import java.util.Objects;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+public final class ContractMetadata {
+
+  // Note that we can't use a `com.daml.ledger.javaapi.data.Timestamp` here because
+  // it only supports milliseconds-precision and we require lossless conversions through
+  // from/toProto.
+  public final ByteString driverMetadata;
+  public final com.google.protobuf.Timestamp createdAt;
```

Comment:
On reflection, a better type choice:

```suggestion
  public final Instant createdAt;
```

`TransactionTree#effectiveAt` is an example (also for to/from conversion).

Reply:
changed!